src/Transport/Receiving/MessagePump.cs

```diff
@@ -271,16 +154,80 @@ async Task ReceiveMessage(CancellationToken messageReceivingCancellationToken)
             // need to catch OCE here because we are switching token
             try
             {
-                await ProcessMessage(message, messageId, headers, body, messageProcessingCancellationTokenSource.Token).ConfigureAwait(false);
+                await ProcessMessage(message, arg, messageId, headers, body, messageProcessingCancellationTokenSource.Token).ConfigureAwait(false);
             }
             catch (Exception ex) when (ex.IsCausedBy(messageProcessingCancellationTokenSource.Token))
             {
                 Logger.Debug("Message processing canceled.", ex);
             }
         }
 
-        async Task ProcessMessage(ServiceBusReceivedMessage message, string messageId, Dictionary<string, string> headers, BinaryData body, CancellationToken messageProcessingCancellationToken)
+#pragma warning disable PS0018
+        async Task OnProcessorError(ProcessErrorEventArgs processErrorEventArgs)
+#pragma warning restore PS0018
+        {
+            string message = $"Failed to receive a message on pump '{processErrorEventArgs.Identifier}' listening on '{processErrorEventArgs.EntityPath}' connected to '{processErrorEventArgs.FullyQualifiedNamespace}' due to '{processErrorEventArgs.ErrorSource}'. Exception: {processErrorEventArgs.Exception}";
+            // Making sure transient exceptions do not trigger the circuit breaker.
+            if (processErrorEventArgs.Exception is ServiceBusException { IsTransient: true })
+            {
+                Logger.Debug(message, processErrorEventArgs.Exception);
+                return;
+            }
+
+            Logger.Warn(message, processErrorEventArgs.Exception);
+            await circuitBreaker.Failure(processErrorEventArgs.Exception, processErrorEventArgs.CancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        public Task ChangeConcurrency(PushRuntimeSettings newLimitations, CancellationToken cancellationToken = default)
+        {
+            limitations = newLimitations;
+            processor.UpdateConcurrency(limitations.MaxConcurrency);
```

Comment:
Unfortunately, it looks like when https://github.com/Particular/NServiceBus/pull/6214 was done the tests have only be added to the Core folder which means they are not executed in downstreams. So all transport repositories are essentially flying blind as of today and we need to manually verify this or copy and paste some of the acceptance tests from Core into this repository even though all other transports have the same issue.

Reply:
Previous version did when Starting again adjust the prefetch multipler. I will see if that was just a side effect of the impl choice or whether it was intentionally chosen like that.

I'm happy to roll back otherwise. I would love though to use what the SDK provides.